pyccl/baryons/baccoemu_baryons.py

```diff
@@ -0,0 +1,169 @@
+            emupars = dict(
+                omega_cold=np.full((len(a)),
+                                   cosmo['Omega_c'] + cosmo['Omega_b']),
+                omega_baryon=np.full((len(a)), cosmo['Omega_b']),
+                ns=np.full((len(a)), cosmo['n_s']),
+                hubble=np.full((len(a)), cosmo['h']),
+                neutrino_mass=np.full((len(a)), np.sum(cosmo['m_nu'])),
+                w0=np.full((len(a)), cosmo['w0']),
+                wa=np.full((len(a)), cosmo['wa']),
+                expfactor=a
+            )
+        else:
+            emupars = dict(
+                omega_cold=cosmo['Omega_c'] + cosmo['Omega_b'],
+                omega_baryon=cosmo['Omega_b'],
+                ns=cosmo['n_s'],
+                hubble=cosmo['h'],
+                neutrino_mass=np.sum(cosmo['m_nu']),
+                w0=cosmo['w0'],
+                wa=cosmo['wa'],
+                expfactor=a
+            )
+
+        # if cosmo contains sigma8, we use it for baccoemu, otherwise we pass
+        # A_s to the emulator
+        if np.isnan(cosmo['A_s']):
+            # note that ccl parametrises sigma8 of the total matter power
+            # spectrum while baccoemu defines it in terms of the cdm+baryons
+            # power spectrum; so we have to convert from total to cold sigma8
+            sigma8tot = cosmo['sigma8']
+            sigma8cold = self._sigma8tot_2_sigma8cold(emupars, sigma8tot)
+            if hasattr(a, '__len__'):
+                emupars['sigma8_cold'] = np.full((len(a)), sigma8cold)
+            else:
+                emupars['sigma8_cold'] = sigma8cold
+        else:
+            if hasattr(a, '__len__'):
+                emupars['A_s'] = np.full((len(a)), cosmo['A_s'])
+            else:
+                emupars['A_s'] = cosmo['A_s']
+
+        # baccoemu internally interpolates k with a cubic spline
+        # it returns k, boost, so, since we are already requesting a specific
+        # k-vector we can ignore the first returned object
+        _, fka = self.mpk.get_baryonic_boost(k=k / cosmo['h'],
+                                             **{**emupars, **self.bcm_params})
+
+        return fka
+
+    def update_parameters(self, log10_M_c=None, log10_eta=None,
+                          log10_beta=None, log10_M1_z0_cen=None,
+                          log10_theta_out=None, log10_theta_inn=None,
+                          log10_M_inn=None):
+        """Update BCM parameters. All parameters set to ``None`` will
+        be left untouched.
+        """
+        _kwargs = locals()
+        _new_bcm_params = {key: _kwargs[key] for key in
+                           set(list(_kwargs.keys())) - set(['self'])}
+        new_bcm_params = {}
+        for key in _new_bcm_params:
+            if _new_bcm_params[key] is not None:
+                new_bcm_params[key[6:]] = _new_bcm_params[key]
+        self.bcm_params.update(new_bcm_params)
+
+    def _include_baryonic_effects(self, cosmo, pk):
```

Comment:
It seems this function is not tested at all. Can you add a quick test? Just see that you can apply baryons to a given Pk and that the ratio before and after agrees with the boost factor, for instance.

Reply:
Done